sdl-snowflake/src/main/scala/io/smartdatalake/workflow/dataobject/SnowflakeTableDataObject.scala

```diff
@@ -45,31 +51,35 @@ import net.snowflake.spark.snowflake.Utils.SNOWFLAKE_SOURCE_NAME
  * @param comment      An optional comment to add to the table after writing a DataFrame to it
  * @param metadata     meta data
  */
+// TODO: we should add virtual partitions as for JdbcTableDataObject and KafkaDataObject, so that PartitionDiffMode can be used...
 case class SnowflakeTableDataObject(override val id: DataObjectId,
                                     override var table: Table,
-                                    override val schemaMin: Option[SparkStructType] = None,
+                                    override val schemaMin: Option[GenericSchema] = None,
                                     saveMode: SDLSaveMode = SDLSaveMode.Overwrite,
                                     connectionId: ConnectionId,
                                     comment: Option[String] = None,
                                     override val metadata: Option[DataObjectMetadata] = None)
                                    (@transient implicit val instanceRegistry: InstanceRegistry)
-  extends TransactionalSparkTableDataObject
-    with CanCreateSnowparkDataFrame
-    with CanWriteSnowparkDataFrame {
+  extends TransactionalSparkTableDataObject {
```

Comment:
SnowflakeTableDataObject extends something Spark-Specific?... That sounds wrong

Reply:
No that's correct. The SnowflakeTableDataObject can be used with Spark (Snowflake Spark Connector) and Snowflake (Snowpark).